src/main/java/bgu/spl/mics/application/passiveObjects/Diary.java

```diff
@@ -63,33 +66,25 @@ public void addReport(Report reportToAdd){
 	 * List of all the reports in the diary.
 	 * This method is called by the main method in order to generate the output.
 	 */
-	public void printToFile(String filename) throws IOException {
-		_lock.readLock().lock();
+	public synchronized void printToFile(String filename) throws IOException {
 		Gson gson = new Gson();
 		String reportsToPrint = gson.toJson(_diary);
-		PrintWriter writer = null;
 		Path file = Paths.get(filename);
 		Files.write(file, Collections.singleton(reportsToPrint), StandardCharsets.UTF_8);
-		_lock.readLock().unlock();
 	}
 
 	/**
 	 * Gets the total number of received missions (executed / aborted) be all the M-instances.
 	 * @return the total number of received missions (executed / aborted) be all the M-instances.
 	 */
-	public int getTotal(){
-		return _total.get();
+	public synchronized int getTotal(){
+		return _total;
 	}
 
 	/**
 	 * Increments the total number of received missions by 1
 	 */
-	public void incrementTotal(){
-		int localTotal;
-		do
-		{
-			localTotal = _total.get();
-		}
-		while (!_total.compareAndSet(localTotal, localTotal+1));
+	public synchronized void incrementTotal(){
+		_total++;
```

Comment:
I think this one is better with the AtomicInteger. no need to synchornize here. synchornize only when adding, printing, or getting the diary, and lock on this object..